### PR TITLE
[subset] make glyph-names option match fonttools behavior

### PR DESCRIPTION
**Pull request by qxliu76**
Don't encode psNames that are contained in standard glyph names set

### CODE REVIEW

File: src/hb-ot-standard-glyph-order-static.hh [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+    const char *g = glyph;
+    for (unsigned i = 0; i < key.length; i++)
+    {
+      char k = key.arrayZ[i];
+      if (*g == '\0' || k > *g) return 1;
+      if (k < *g) return -1;
+      g++;
+    }
+    if (*g == '\0') return 0;
+    return -1;
```
Comment (behdad) [inline src/hb-ot-standard-glyph-order-static.hh:45]: Why are you not using `strcmp`?!

Comment (qxliu76) [reply src/hb-ot-standard-glyph-order-static.hh:45]: oh, I found arrayZ in hb_bytes_t doesn't include a trailing '\0'. I will change to strcmp

Comment (qxliu76) [reply src/hb-ot-standard-glyph-order-static.hh]: done

**Comment by behdad** [comment]
Thanks. See if you can use `hb-string-array` instead. That should save a couple kb.

**Comment by qxliu76** [comment]
> Thanks. See if you can use `hb-string-array` instead. That should save a couple kb.

Done, thank you! I should use already existing utilities. :)

**Comment by behdad** [comment]
Cool. I thought we might have that name list already. :)

File: test/subset/meson.build [hunk 2]
```diff
@@ -36,6 +36,7 @@ tests = [
   'colr_with_components',
   'cbdt',
   'variable',
+  'glyph_names',
```
Comment (khaledhosny) [inline test/subset/meson.build:39]: You need to add this to the relevant `Makefile.sources` to fix the distcheck failure.

Comment (qxliu76) [reply test/subset/meson.build:39]: Done, thank you!

**Pull request merged**